app/uk/gov/hmrc/agentstatuschange/controllers/AuthActions.scala

```diff
@@ -57,15 +60,37 @@ trait AuthActions extends AuthorisedFunctions {
         body(id)
       }
 
-  def onlyStride(strideRole: String)(action: => Credentials => Future[Result])(
+  val basicAuthHeader: Regex = "Basic (.+)".r
+  val decodedAuth: Regex = "(.+):(.+)".r
+
+  private def decodeFromBase64(encodedString: String): String =
```

Comment:
you can move the above three lines into AuthActions so we can reuse them in unit tests or anywhere we want ?

Reply:
What do you mean by AuthActions object?